File: modularity_conn/44.voxel_module_setmembers.py
#!usr/bin/python

"""
Find whether voxels part of a module in one condition prune or reorganize
"prune" -> Partition big module into sub-modules
"reorganize" -> different connections, voxels are that were together in one module are now in different modules
"""

import os
from glob import glob

conditions = range(1,5)
subjects = ["ANGO","MYTP","TRCO","PIGL","SNNW","LDMW","FLTM","EEPA","DNLN","CRFO","ANMS","MRZM","MRVV","MRMK","MRMC","MRAG","MNGO","LRVN","CLFR"]

for ss in subjects:
    os.chdir(os.environ["state"]+"/"+ss+"/corrTRIM_BLUR/")
    print os.getcwd()
    f3 = glob("cleanTS.3.*justcomm")
    fl = glob("cleanTS.1.*justcomm")
    com3f = open(''.join(f3))
    com1f = open(''.join(f1))
    link3f_name = "/mnt/tier2/urihas/Andric/steadystate/graymask_links/"+ss+".3.links_corval"
    link1f_name = "/mnt/tier2/urihas/Andric/steadystate/graymask_links/"+ss+".1.links_corval"
    link3f = open(link3f_name)
    link1f = open(link1f_name)

    """
    get module identifiers
    """
    comm_array = []
    for line in inputf:
        comm_array.append([int(x) for x in line.split()][0])

    """
    get links
    """
    link3_pair = []
    for line in link3f:
        link3_pair.append((int(line.split()[0]), int(line.split()[1])))

    link1_pair = []
    for line in link1f:
        link1_pair.append((int(line.split()[0]), int(line.split()[1])))

    """
    See if link in condition3 if in condition1 too
    """
    res = [rec for rec in set(link3_pair) if rec in set(link1_pair)] 
    

    """
    Now see if voxels link within module
    """
    same_comm = []
    link_both = []
    for line in link3_pair:
        if comm_array[int(link3_pair[0][0])] == comm_array[int(link3_pair[0][1])]: ## within-module filter: finds 'if' the community IDs are same for the two voxels that make the link
            for i in link3_pair:
                for v in link1_pair:
                    if i == v and if comm_array[int(link1_pair[0][0])] == comm_array[int(link1_pair[0][1])]: ## 'if' the link also in condition 1 AND if they still in the same community
                        link_both.append(i)

            same_comm.append("YES")
            if comm_array[int(link1_pair[0][0])] == comm_array[int(link1_pair[0][1])]:
                link_both.append(link_pair[]) 
        else:
            same_comm.append("NO")


    """
    See if voxel pair also in Highly ordered condition
    """

res = [rec for rec in link3_pair if rec in link1_pair] ## see below
[rec for rec in t33 if rec in t11] ## this gives the 'rec' in a list (t33) if it's also in (t11)
##[i for i, v in enumerate(t3) if v[0] == '1' and v[1] == '212']
##[e for e in comm_array if e==3]

File: modularity_conn/44.voxel_module_setmembers.R
## characterize voxels by their module membership between Highly ordered and Random conditions
subjects <- c("ANGO","CLFR","MYTP","TRCO","PIGL","SNNW","LDMW","FLTM","EEPA","DNLN","CRFO","ANMS","MRZM","MRVV","MRMK","MRMC","MRAG","MNGO","LRVN")
conditions <- seq(4)
cutoff = 100

average_counts <- c()

for (ss in subjects)
{
    setwd(paste("/mnt/tier2/urihas/Andric/steadystate/",ss,"/corrTRIM_BLUR",sep=""))
    dat1 <- as.matrix(read.table(list.files(pattern=paste("cleanTS.1.*.justcomm",sep="")))) 
    dat1 <- cbind(c(1:length(dat1)), dat1)
    dat3 <- as.matrix(read.table(list.files(pattern=paste("cleanTS.3.*.justcomm",sep=""))))
    dat3 <- cbind(c(1:length(dat3)), dat3)
    ordered_agframe1 = aggregate(dat1[,2], list(dat1[,2]), length)[order(-aggregate(dat1[,2], list(dat1[,2]), length)$x),]
    ordered_agframe3 = aggregate(dat3[,2], list(dat3[,2]), length)[order(-aggregate(dat3[,2], list(dat3[,2]), length)$x),]

## Does disorder partition modules that exist in "Highly ordered" (pruning)
## Or do voxels form new setsi, i.e., which voxels connect is different (re-organization)
## see if in Random condition voxels in a set are also in a set in Highly Ordered 

    for (i in unique(dat1[,2]))
    {
        which(dat1[,1]set1 %in% dat3[,1]set1)


set1 = subset(dat1, dat1[,2]==ordered_agframe1$Group.1[1])[,1]
set2 = subset(dat1, dat1[,2]==ordered_agframe1$Group.1[2])[,1]
set1_3 = subset(dat3, dat3[,2]==ordered_agframe3$Group.1[1])[,1]

subset(dat1, dat1[,2]==unique(dat1[,2])[1])[,1]
ordered_agframe = aggregate(dat1[,2], list(dat1[,2]), length)[order(-aggregate(dat1[,2], list(dat1[,2]), length)$x),]
length(intersect(set1,set1_3))


